Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/ValidationServiceRequestProcessor.kt

```diff
@@ -0,0 +1,109 @@
+package de.rki.coronawarnapp.dccticketing.core.service.processor
+
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingErrorCode
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException
+import de.rki.coronawarnapp.dccticketing.core.server.DccTicketingServer
+import de.rki.coronawarnapp.dccticketing.core.server.getServiceIdentityDocument
+import de.rki.coronawarnapp.dccticketing.core.transaction.DccJWK
+import de.rki.coronawarnapp.dccticketing.core.transaction.DccTicketingService
+import de.rki.coronawarnapp.dccticketing.core.transaction.DccTicketingServiceIdentityDocument
+import timber.log.Timber
+import javax.inject.Inject
+
+class ValidationServiceRequestProcessor @Inject constructor(
+    private val dccTicketingServer: DccTicketingServer
+) {
+
+    private val regexRSAOAEPWithSHA256AESCBC = """/ValidationServiceEncScheme-RSAOAEPWithSHA256AESCBC${'$'}/"""
+        .toRegex()
+
+    private val regexRSAOAEPWithSHA256AESGCM =
+        """/ValidationServiceEncScheme-RSAOAEPWithSHA256AESGCM${'$'}/"""
+            .toRegex()
+
+    @Throws(DccTicketingException::class)
+    suspend fun requestValidationService(
+        validationService: DccTicketingService,
+        validationServiceJwkSet: Set<DccJWK>
+    ): Output {
+        Timber.d(
+            "requestValidationService(validationService=%s, validationServiceJwkSet=%s)",
+            validationService,
+            validationServiceJwkSet
+        )
+
+        // 1. Call Service Identity Document
+        val serviceIdentityDocument = dccTicketingServer.getServiceIdentityDocument(
+            url = validationService.serviceEndpoint,
+            parserErrorCode = DccTicketingErrorCode.VS_ID_PARSE_ERR,
+            clientErrorCode = DccTicketingErrorCode.VS_ID_CLIENT_ERR,
+            serverErrorCode = DccTicketingErrorCode.VS_ID_SERVER_ERR,
+            noNetworkErrorCode = DccTicketingErrorCode.VS_ID_NO_NETWORK
+        )
+
+        // Checking the Server Certificate Against a Set of JWKs
+        // TODO: Add cert check
```

Comment:
maybe not here but rather as `CertificatePinner` implementation injected into `OkHttpClient`

Reply:
it's also OK to do in as separate PR